### ISSUE TITLE
Exceptions when used with Scalaz (Unicode in method names)

### ISSUE THREAD

**Issue by kstep** [ensime/ensime-vim#182]
Using with YCM via omnifunc integration.
Trying to complete the following line:
```scala
import scalaz.std.option.
```

Got an exception:
```
Error detected while processing function youcompleteme#Complete..<SNR>129_GetCompletions..provider#python#Call:
line   18:
UnicodeDecodeError('ascii', '\x92', 0, 1, 'ordinal not in range(128)')
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/neovim/msgpack_rpc/session.py", line 177, in handler
    rv = self._request_cb(name, args)
  File "/usr/lib/python2.7/site-packages/neovim/api/common.py", line 237, in filter_request_cb
    walk(self._in, args, self, name, 'request'))
  File "/usr/lib/python2.7/site-packages/neovim/plugin/host.py", line 65, in _on_request
    rv = handler(*args)
  File "/usr/share/nvim/runtime/autoload/provider/script_host.py", line 77, in python_execute
    exec(script, self.module.__dict__)
  File "<string>", line 1, in <module>
UnicodeDecodeError: 'ascii' codec can't decode byte 0x92 in position 0: ordinal not in range(128)
Error detected while processing function youcompleteme#Complete..<SNR>129_GetCompletions..provider#python#Call:
```

**Comment by jvican** [comment]
This is probably a problem of Python compatibilities, see [this](https://stackoverflow.com/questions/26619801/unicodedecodeerror-ascii-codec-cant-decode-byte-0x92-in-position-47-ordinal).
  
I'll take a look at this tonight. I need your version of Python and it would be very advantageous if you could tell me which ensime and ensime-vim version you're using.

**Comment by ChrisCoffey** [comment]
is this failing in YCM's python code or ensime's? From the backtrace it looks like YCM is the source of the error.

**Comment by kstep** [comment]
@jvican:
```
 — ~ pts/7: python --version
Python 3.5.1
 — ~ pts/7: python2 --version
Python 2.7.11
 — ~ pts/7: uname -a
Linux workbook 4.4.3-1-ARCH #1 SMP PREEMPT Fri Feb 26 15:09:29 CET 2016 x86_64 GNU/Linux
 — ~ pts/7: nvim --version
NVIM 0.1.2
Build type: RelWithDebInfo
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -Wconversion -O2 -g -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -I/build/neovim/src/build/config -I/build/neovim/src/neovim-0.1.2/src -I/usr/include -I/usr/include -I/usr/include/luajit-2.0 -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/build/neovim/src/build/src/nvim/auto -I/build/neovim/src/build/include
Compiled by builduser

Optional features included (+) or not (-): +acl   +iconv    +jemalloc
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"
```

ensime-vim from master branch at 1ad857704069e9bed59ed43f5c26c81e60520316.
ensime-server from [AUR](https://aur.archlinux.org/packages/ensime-server/) 0.9.10-5.

@ChrisCoffey it's possible, though I'm not 100% sure for now.

**Comment by jvican** [comment]
Could you please give me the Python version that Neovim uses?

**Comment by kstep** [comment]
Obviously 2.7.11.

**Comment by jvican** [comment]
Why would it be obvious?
  
Python3 is officially supported and you have both versions installed. https://github.com/neovim/neovim/pull/2208

**Comment by kstep** [comment]
@jvican thanks for the pointers!

It seemed obvious enough from logs (File "**/usr/lib/python2.7**/site-packages/neovim/plugin/host.py") that python2 was used. But after your comment I poked around, and made sure YCM is built for python3. The problem was, although YCM was really built for python3, it was using python2 to run things internally (with `py` vim command instead of `py`). I had to change YCM sources to force it use py3 (`autoload/youcompleteme.vim`):

```viml
function! s:UsingPython2()
  " I'm willing to bet quite a bit that sooner or later, somebody will ask us to
  " make it configurable which version of Python we use.
  if has('python') " <-- removed these three lines
    return 1
  endif
  return 0
endfunction
```

Not it works like a charm!
